packages/hdwallet-trezor/src/trezor.ts

```diff
@@ -44,6 +49,115 @@ export function isTrezor(wallet: HDWallet): wallet is TrezorHDWallet {
   return typeof wallet === 'object' && wallet._isTrezor === true
 }
 
+function describeETHPath (path: BIP32Path): PathDescription {
+  let pathStr = addressNListToBIP32(path)
+  let unknown: PathDescription = {
+    verbose: pathStr,
+    coin: 'Ethereum',
+    isKnown: false
+  }
+
+  if (path.length != 5)
+    return unknown
+
+  if (path[0] != 0x80000000 + 44)
+    return unknown
+
+  if (path[1] != 0x80000000 + slip44ByCoin('Ethereum'))
+    return unknown
+
+  if (path[2] !== 0x80000000)
+    return unknown
+
+  if (path[3] != 0)
+    return unknown
+
+  if ((path[4] & 0x80000000) !== 0)
+    return unknown
+
+  let accountIdx = path[4] & 0x7fffffff
+  return {
+    verbose: `Ethereum Account #${accountIdx}`,
+    coin: 'Ethereum',
+    accountIdx,
+    wholeAccount: true,
+    isKnown: true
+  }
+}
+
+function describeUTXOPath (path: BIP32Path, coin: Coin, scriptType: BTCInputScriptType) {
+  let pathStr = addressNListToBIP32(path)
+  let unknown: PathDescription = {
+    verbose: pathStr,
+    coin,
+    scriptType,
+    isKnown: false
+  }
+
+  if (!Btc.btcSupportsCoin(coin))
+    return unknown
+
+  if (!Btc.btcSupportsScriptType(coin, scriptType))
+    return unknown
+
+  if (path.length !== 3 && path.length !== 5)
+    return unknown
+
+  if ((path[0] & 0x80000000) >>> 0 !== 0x80000000)
+    return unknown
+
+  let purpose = path[0] & 0x7fffffff
+
+  if (![44, 49, 84].includes(purpose))
```

Comment:
Do we have these purposes defined somewhere like an enum?

Reply:
They're self-documenting.

https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki
https://github.com/bitcoin/bips/blob/master/bip-0049.mediawiki
https://github.com/bitcoin/bips/blob/master/bip-0084.mediawiki